### ISSUE TITLE
Using != in a select query statement may cause some data to be missed.

### ISSUE THREAD

**Issue by Kyle-Derrick** [ClickHouse/ClickHouse#56191]
clickhouse version: 21.3.6.55
table info:
``` sql
CREATE TABLE IF NOT EXISTS test (
-- ...
    `time` UInt32
)
ENGINE = MergeTree
PARTITION BY toStartOfDay(toDateTime(`time`))
ORDER BY `time`
```

The following SQL statement returns an empty result: 

```sql
select * from test where `time` != 1698825461 and `time` >= 1698768000 and `time` <= 1698854399
```

The following SQL statement executes normally:

```sql
select * from test where 0=(`time` = 1698825461) and `time` >= 1698768000 and `time` <= 1698854399
```

**Comment by den-crane** [comment]
21.3 is out of support.
This bug has been fixed two years ago.
You need to upgrade, use 23.8

**Comment by den-crane** [comment]
It's even fixed in 21.3

21.3.6.55 https://fiddle.clickhouse.com/624e7d15-e403-459e-b645-bbac51353d74
21.3.20.1 https://fiddle.clickhouse.com/aee86e1d-efc1-4d11-b7c5-f6fac4799ab3
